api/server/sdk/volume_ops.go

```diff
@@ -609,6 +677,11 @@ func (s *VolumeServer) Update(
 	if err != nil {
 		return nil, err
 	}
+
+	// avoid side effect while applying with stale config by masking
+	// other parts of the spec.
+	maskUnModified(updatedSpec, req.GetSpec())
+
 	// Send to driver
 	if err := s.driver(ctx).Set(req.GetVolumeId(), locator, updatedSpec); err != nil {
```

Comment:
This ends up calling porx repo: storage/driver/volume/proto_driver.go
https://github.com/portworx/porx/blob/e0bd8dab2f2f684c4b06f7e55ef862ea63d02f7a/storage/driver/volume/proto_driver.go#L5710